src/FluentResults/Results/Result.cs

```diff
@@ -148,6 +148,56 @@ public Result<TNewValue> ToResult<TNewValue>(Func<TValue, TNewValue> valueConver
                 .WithReasons(Reasons);
         }
 
+        /// <summary>
```

Comment:
You didn't added the Bind method to the Result class (without value). Why? 
I think the Bind() method could also be useful on this class

```
Result.Ok()
    .Bind(SomeMethod)
    .Bind(SomeAnotherMethod);
```

Reply:
I didn't because there is no value to bind. However, the method can be implemented, its `bind` can be `Func<Result<T>>`, with the same rules. This will allow to depend on the state of the previous result. Also I will add `BindAsync` as well.

Reply:
Added the methods